### PR TITLE
Chore: refactor vela cli entrance

### PR DESCRIPTION
**Pull request by Somefive**
### Description of your changes

1. Refactor vela cli entrance. Reorder command groups and hide part of the commands. (HIDE means the command can be still used but not displayed on the help page.)
2. Fix incorrect string based order sort.
3. Refactor short description.

```shell
$ vela
A Highly Extensible Platform Engine based on Kubernetes and Open Application Model.

Getting Started:
  init                  Create scaffold for an application.                               
  up                    Deploy one application.                                           
  status                Show status of an application.                                    
  ls                    List applications.                                                
  delete                Delete an application.                                            
  env                   Manage environments for vela applications to run.                 
  show                  Show the reference doc for a component, trait, policy or workflow.

Managing Applications:
  addon                 Manage addons for extension.                                             
  dry-run               Dry Run an application, and output the K8s resources as result to stdout.
  live-diff             Compare application and revisions.                                       
  logs                  Tail logs for application.                                               
  port-forward          Forward local ports to container/service port of vela application.       
  exec                  Execute command in a container.                                          
  revision              Manage Application Revisions.                                            
  debug                 Debug running application.                                               

Continuous Delivery:
  workflow              Operate application delivery workflow.
  adopt                 Adopt resources into new application. 

Managing Platform:
  top                   Launch UI to display the platform overview.
  cluster               Manage Kubernetes clusters.                
  auth                  Manage identity and authorizations.        
  config                Manage the configs.                        
  config-template       Manage the template of config.             

Managing Extensions:
  uischema              Manage UI schema for addons.
  component             List/get components.        
  trait                 List/get traits.            
  def                   Manage definitions.         

System Tools:
  install               Installs or Upgrades Kubevela control plane on a Kubernetes cluster.
  uninstall             Uninstalls KubeVela from a Kubernetes cluster.                      
  system                Manage system.                                                      
  version               Prints vela build version information.                              

Auxiliary Tools:
  kube                  Managing native Kubernetes resources across clusters.              
  cuex                  Manage CueX engine for compile.                                    
  ql                    Show result of executing velaQL.                                   
  completion            Output shell completion code for the specified shell (bash or zsh).
  help                  Help about any command.                                            

Flags:
  -h, --help   help for vela

Use "vela [command] --help" for more information about a command.
```

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: references/cli/provider.go [hunk 1]
```diff
@@ -34,11 +34,11 @@ import (
...
 	cmd := &cobra.Command{
 		Use:   "provider",
-		Short: "Authenticate Terraform Cloud Providers",
-		Long:  "Authenticate Terraform Cloud Providers by managing Terraform Controller Providers with its credential secret",
+		Short: "Authenticate terraform cloud providers.",
+		Long:  "Authenticate terraform cloud providers by managing terraform controller providers with its credential secret.",
 		Annotations: map[string]string{
 			types.TagCommandOrder: order,
-			types.TagCommandType:  types.TypeExtension,
+			types.TagCommandType:  types.TypePlatform,
```
Comment (chivalryq) [inline references/cli/provider.go:41]: I think provider command can be legacy. New `config` is a replacement.

Comment (Somefive) [reply references/cli/provider.go]: Done

File: references/cli/show.go [hunk 2]
```diff
@@ -121,7 +121,8 @@ func NewCapabilityShowCommand(c common.Args, ioStreams cmdutil.IOStreams) *cobra
 			return ShowReferenceConsole(ctx, c, ioStreams, capabilityName, namespace, location, i18nPath, int64(ver))
 		},
 		Annotations: map[string]string{
-			types.TagCommandType: types.TypeStart,
+			types.TagCommandType:  types.TypeLegacy,
```
Comment (wonderflow) [inline references/cli/show.go:124]: please keep it

File: references/cli/env.go [hunk 3]
```diff
@@ -37,7 +37,7 @@ func NewEnvCommand(c common.Args, order string, ioStream cmdutil.IOStreams) *cob
 		Long:                  "Manage environments for vela applications to run.",
 		Annotations: map[string]string{
 			types.TagCommandOrder: order,
-			types.TagCommandType:  types.TypeStart,
+			types.TagCommandType:  types.TypeLegacy,
```
Comment (wonderflow) [inline references/cli/env.go:40]: please keep it

**Review by wonderflow — CHANGES_REQUESTED** [review]
1. `vela env` is used in our first quick start demo, we must keep it unless we remove that in the quick start.
2. `vela show` is useful, this don't add any burden.

File: references/cli/show.go (hunk 2 above)
Comment (Somefive) [reply references/cli/show.go]: Done

File: references/cli/env.go (hunk 3 above)
Comment (Somefive) [reply references/cli/env.go]: Done

**Review by wonderflow — APPROVED** [review]

**Review by chivalryq — APPROVED** [review]

**Pull request merged**